### PR TITLE
chore(types): IgnorePlugin

### PR DESCRIPTION
**Pull request by byzyk**
**What kind of change does this PR introduce?**

types

**Did you add tests for your changes?**

n/a

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

n/a

### CODE REVIEW

File: lib/IgnorePlugin.js [hunk 1]
```diff
@@ -34,12 +45,12 @@ class IgnorePlugin {
...
-	/*
-	 * Returns true if result should be ignored.
-	 * false if it shouldn't.
-	 *
-	 * Not that if "contextRegExp" is given, both the "resourceRegExp"
+	/**
+	 * Note that if "contextRegExp" is given, both the "resourceRegExp"
 	 * and "contextRegExp" have to match.
+	 *
+	 * @param {TODO} result result
```
Comment (byzyk) [inline lib/IgnorePlugin.js]: Not sure what exactly to put here, would appreciate some guidance

File: lib/IgnorePlugin.js [hunk 2]
```diff
@@ -50,6 +61,10 @@ class IgnorePlugin {
 		);
 	}
 
+	/**
+	 * @param {TODO} result result
+	 * @returns {TODO|null} returns result or null if result should be ignored
```
Comment (byzyk) [inline lib/IgnorePlugin.js]: Same as above

File: lib/IgnorePlugin.js (hunk 1 above)
Comment (byzyk) [reply lib/IgnorePlugin.js]: I see it depends on #7299

Comment (byzyk) [reply lib/IgnorePlugin.js]: After digging a bit deeper into it turns out this depends on `tapable` typing, or am I missing something?

Are we planning to add types in `tapable` as well or shall we leave this as `TODO` for the time being?

/cc @sokra @ooflorent @mohsen1

**Comment by sokra** [comment]
Thanks

**Pull request merged**